okhttp/src/main/java/okhttp3/Headers.java

```diff
@@ -199,6 +195,26 @@ public static Headers of(Map<String, String> headers) {
     return new Headers(namesAndValues);
   }
 
+  private static void checkNameAndValue(String name, String value) {
+    if (name == null) throw new IllegalArgumentException("name == null");
+    if (name.isEmpty()) throw new IllegalArgumentException("name is empty");
+    for (int i = 0, length = name.length(); i < length; i++) {
+      char c = name.charAt(i);
+      if (c <= '\u001f' || c >= '\u007f') {
+        throw new IllegalArgumentException(String.format(
+            "Unexpected char %#04x at %d in header name: %s", (int) c, i, name));
+      }
+    }
+    if (value == null) throw new IllegalArgumentException("value == null");
+    for (int i = 0, length = value.length(); i < length; i++) {
+      char c = value.charAt(i);
+      if (c <= '\u001f' || c >= '\u007f') {
+        throw new IllegalArgumentException(String.format(
+            "Unexpected char %#04x at %d in %s value: %s", (int) c, i, name, value));
```

Comment:
This line is the only one I actually care about... I just want to see the header name in this error message.